### ISSUE TITLE
Enable background synchronization by default

### ISSUE THREAD

**Issue by stefan-niedermann** [stefan-niedermann/nextcloud-notes#1168]
- Enable background synchronization by default for new users
- Enable background synchronization in a migration for all users which have not set explicitly a state yet
- *Optional*: Replace the options with a toggle switch and only provide `On` and `Off`. Use 15 minutes. Since we only use it on Wi-Fi, this won't consume much traffic and the impacts on the battery life is insignificant

**Comment by Cui-Yusong** [comment]
I would like to ask whether the solution to this problem is as follows:
In the original case, background synchronization is not enabled by default.
The change that needs to be made is to enable background synchronization for new users by default.Then add a UI that provides the ON OFF option instead of options.
Is it correct? Thanks very much.

**Comment by MasterWanna** [comment]
I have tried some methods, but they does not work. I have a question that where can I set default value for the variable of `background synchronization` when it is just installed?

**Comment by stefan-niedermann** [comment]
For existing installation we will need a migration. Increment the database version (`NotesDatabase.java`) and add a new `Migration`. There are already migrations which only manipulate the `SharedPreferences`, you can recognize them often on having a `context` parameter.

**Comment by MasterWanna** [comment]
I find that, and all preset values should be set as `Migration`, and to implement this task, I need to add a new `Migration`, is it right?

**Comment by MasterWanna** [comment]
I found a solution which can achieve this, by editing `defaultValue` in `src\main\res\xml\preferences.xml`
![QQ图片20210502150224](https://user-images.githubusercontent.com/70736568/116805209-734f1880-ab57-11eb-984b-f3db93399eac.png)
Is it a avaliable for this?

**Comment by stefan-niedermann** [comment]
Yes, this will work. Though we will still need a migration, because if an user changed the default value in an old version, we need to respect this and migrate the value (`off` or a time interval value) to the new boolean setting.

However, all of this stuff is implementation details, so please continue discussion in a Pull Request. Issues are a matter of "What", Pull Requests are a matter of "How" to implement. 🙂

**Comment by MasterWanna** [comment]
Thanks, got it!
